dashboard/app/controllers/api_controller.rb

```diff
@@ -11,6 +11,51 @@ def user_hero
     head :not_found unless current_user
   end
 
+  def update_lockable_state
+    updates = params.require(:updates)
+    updates.to_a.each do |item|
+      UserLevel.find(item[:user_level_id]).update(
+        submitted: item[:locked],
+        view_answers: item[:view_answers],
+        unlocked_at: item[:locked] ? nil : Time.now
+      )
+    end
+    render json: {}
+  end
+
+  # For a given users, gets the lockable state for each user in each of their sections
+  def lockable_state
+    return unless current_user
+
+    data = current_user.sections.each_with_object({}) do |section, section_hash|
+      @section = section
+      load_script
+      section_hash[section.id] = {
+        section_id: section.id,
+        section_name: section.name,
+        stages: @script.stages.each_with_object({}) do |stage, stage_hash|
+          next unless stage.lockable
+          # assumption that lockable stages have a single (assessment) level
+          if stage.script_levels.length > 1
+            raise 'Expect lockable stages to have a single script_level'
+          end
+          script_level = stage.script_levels[0]
+          stage_hash[stage.id] = section.students.map do |student|
+            user_level = UserLevel.find_or_create_by(user_id: student.id, level: script_level.level, script_id:  @script.id)
+            {
+              user_level_id: user_level.id,
+              name: student.name,
+              locked: user_level.locked?,
+              view_answers: user_level.view_answers
+            }
+          end
+        end
+      }
+    end
+
+    render json: data
+  end
+
```

Comment:
Is api_controller the right place for this logic to live? If not, where would it make more sense?